### DISCUSSION TITLE
Publishing Hugo site via GitHub Actions - Alan Pope's blog

### DISCUSSION THREAD

**Discussion by giscus** [popey/popey.com-blog#40] [General]
# Publishing Hugo site via GitHub Actions - Alan Pope's blog

My blog at popey.com/blog is hosted on a Bitfolk VPS, built from the Hugo source code in a public GitHub repo.
My workflow for publishing a post goes like this:
💻 Use whatever machine I’m sat at 🔽 Clone the repo 🗒 Add a new page, edit until ready 🤠 Push directly to the main branch Early on in my use of Hugo, I was manually using hugo and rsync over SSH directly on the VPS.

https://popey.com/blog/2023/09/publishing-hugo-site-via-github-actions/

**Comment by grifferz** [comment]
The step where github SSHes to popey.com, does it not ask about host key every time as the container will not know the identity of popey.com? I've never used GitHub actions but I would expect you to have had to tell SSH to not do host key checking.

**Reply by popey** [reply]
Nope. If you look at the logs from the actions (I assume you can see them) it never asks.

**Reply by grifferz** [reply]
I saw! But I don't understand how. 😀
Does the default SSH client config in a GitHub action disable host key checking?

On the SSH command line that would be:

```
$ ssh -o "StrictHostKeyChecking=no" …
```

Or in `~/.ssh/config` (or `/etc/ssh/ssh_config`) for every host it would be:

```
Host *
    StrictHostKeyChecking no
```

I can see you don't set that anywhere, so I'm wondering how it happens.

**Reply by popey** [reply]
It bodges it into known_hosts every time the action runs.

**Comment by grifferz** [comment]
Oh yes it is right there, sorry. You even explained what it was doing. I skipped both parts because I thought, "oh this is just the part about the user's ssh key". Oops.

Anyway I expect it is not unclear for anyone, but just in case, this command:

```
ssh-keyscan -H ${{ secrets.DEPLOY_HOST_IP }} >> ~/.ssh/known_hosts
```

…does exactly the same as what you would do interactively when you say that you've verified that the thing you're connecting to really is the thing you're connecting to.

As an aside, I did once have someone phone me up to ask if an SSH host key that they were seeing on their screen was the same as the real key of my host.

I wonder if DNSSEC and SSHFP record checking works from within a GitHub Action…
https://dataswamp.org/~solene/2023-08-05-sshfp-dns-entries.html
